### PR TITLE
[choe] replace scraperhelper.NewScraperWithComponentType to scraperhelper.NewScraper as deprecated

### PR DESCRIPTION
**Pull request by mrasu**
To change an argument of `scraperhelper.NewScraper`, `scraperhelper.NewScraperWithComponentType` is introduced in previous version and deprecated in newer version at https://github.com/open-telemetry/opentelemetry-collector/pull/11159.

This PR update dependencies by `multimod sync` and replace `scraperhelper.NewScraperWithComponentType` to `scraperhelper.NewScraper`

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector/issues/9473

### CODE REVIEW

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.22.0
 
-toolchain go1.22.7
+toolchain go1.23.1
```
Comment (crobert-1) [inline cmd/otelcontribcol/go.mod:7]: We shouldn't update the toolchain version here. (Same comment for other toolchain updates)

File: cmd/githubgen/go.mod [hunk 2]
```diff
@@ -4,11 +4,13 @@ go 1.22.0
...
 require (
 	github.com/google/go-github/v64 v64.0.0
-	go.opentelemetry.io/collector/confmap v1.15.1-0.20240911162712-6c2697c4453c
-	go.opentelemetry.io/collector/confmap/provider/fileprovider v1.15.1-0.20240911162712-6c2697c4453c
+	go.opentelemetry.io/collector/confmap v1.15.1-0.20240916093651-6029f3176705
+	go.opentelemetry.io/collector/confmap/provider/fileprovider v1.15.1-0.20240916093651-6029f3176705
 	gopkg.in/yaml.v3 v3.0.1
 )
 
+require github.com/rogpeppe/go-internal v1.12.0 // indirect
```
Comment (crobert-1) [inline cmd/githubgen/go.mod:12]: Can you share why this dependency is required? I wouldn't have expected a new dependency to be required when the only change is updating existing dependencies, but one of them must require it now?

**Comment by mrasu** [comment]
@crobert-1 
Sorry. I ran `make update-otel` but encountered errors saying `go.opentelemetry.io/collector/internal/localhostgate` was not found during gotidy.
So, I fixed this by running `go mod edit -require go.opentelemetry.io/collector/config/internal@v0.109.1-0.20240916093651-6029f3176705` and then executed the commands after gotidy in `update-otel` manually.
However, it made huge diffs including unrelated changes as you mentioned.

So, I recreated the changes by running `multimod sync -s=true -o ../opentelemetry-collector -m beta,stable --commit-hash 6029f31767` and fixed the errors manually.

**Comment by mx-psi** [comment]
Overlaps with #35215, I think @codeboten wants to merge #35215 first

**Comment by codeboten** [comment]
sorry @mrasu! i hadnt seen this PR before i started working on my own update core PR to include the changes i made to the processor helper... i think it's almost ready to merge

**Comment by mrasu** [comment]
@codeboten No worries! close this

**Pull request closed without merging**